sdk/src/client/api/block_builder/input_selection/requirement/amount.rs

```diff
@@ -153,6 +155,22 @@ impl AmountSelection {
             self.inputs_sum += input.output.amount();
             self.newly_selected_inputs.insert(*input.output_id(), input.clone());
 
+            if input.output.native_token().is_some() {
+                // Recalculate the remaining amount, as a new native token may require a new remainder output.
```

Comment:
Will this also select more inputs to cover the possibly higher output amount? And what happens if those inputs need even more remainders?

Reply:
It will increase the required amount, so a bit further down `if self.missing_amount() == 0 {` will not be 0 and thus search for more inputs to cover the required amount
If more inputs are selected with other native tokens, this will just be called again and the required amount will just increase again by a new basic output remainder with a native token